### ISSUE TITLE
Get the cleaning steps to work

### ISSUE THREAD

**Issue by lindsayplatt** [lindsayplatt/state-of-freshwater#10]
With as much data as we have, we are breaking the default `p3_wqp_data_aoi_clean` target due to size limitations - it fails on memory issues, perhaps because it is trying to detect duplicates across too many records at once.

**Comment by lindsayplatt** [comment]
I'm currently testing out an approach where I group by the parameter groups _before_ the cleaning target. Seeing some success so far because one of the branches just completed, but the first parameter (`calcium` - 4.1 mil records) alone took 38 minutes.

**Comment by lindsayplatt** [comment]
The second parameter (`chloride` - 3.1 mil records) took 41 minutes!

**Comment by lindsayplatt** [comment]
The third one (`conductivity` - 14.1 mil records) kept running out of memory. So, I'll be trying to group specifically by the `characteristicNames` this time. That's more groups, but I think it will solve the memory issue. The only other idea I have is to switch some of the code over to `{data.table}`

**Comment by lindsayplatt** [comment]
Hmmm the characteristic `Specific conductance` has 11 million records by itself, so I may need to try `{data.table}`.

```
# A tibble: 8 × 2
  CharacteristicName                                                n
  <chr>                                                         <int>
1 Alkalinity, total                                           1754625
2 Calcium                                                     2376760
3 Chloride                                                    3089597
4 Conductance                                                     181
5 Conductivity                                                2624661
6 Specific conductance                                       11496355
7 Specific Conductance, Calculated/Measured Ratio               34503
8 Specific conductivity***retired***use Specific conductance     1875
```

**Comment by lindsayplatt** [comment]
Current plan: adjust the `add_download_groups()` function to work in `3_harmonize` and add cleaning groups based on `CharacteristicName` but also a max of 1 million records per group. Write out files and then do cleaning by file. Potentially parallelize the cleaning steps on individual files? 

I think to parallelize, you simple run `tar_make_clustermq()` (once you have it installed) instead of `tar_make()`.

**Comment by lindsayplatt** [comment]
Closed by #14
